trio/_core/_windows_cffi.py

```diff
@@ -327,13 +517,14 @@ def raise_winerror(
     filename: str | None = None,
     filename2: str | None = None,
 ) -> NoReturn:
+    assert sys.platform == "win32"  # this should work starting next mypy release
```

Comment:
oh, TIL! I was struggling with this in #2774

Reply:
Yea, I believe this falls under https://github.com/python/mypy/pull/15386

Reply:
yeah I suspect that's a different fix